### ISSUE TITLE
Specify custom order for sidebar

### ISSUE THREAD

**Issue by GordonSmith** [jannikbuschke/gatsby-antd-docs#1]
Is there anyway to specify the order of the sidebar items?

Currently the only way I seem to be able to do this is to:
* the docs/folders to order the "parent" sidebar items.
* Rename the "title" to include a `01` `02` prefix...

**Comment by wagnerjt** [comment]
Hey Gordon, I just found this repo today actually and want to contribute some items I am going to run into digging through it, and this is one of them. 
Can you help me understand if you need to sort the children in some order you choose, or is the children okay to remain sorted in  alphabetical order based on the the markdown Metadata title?

**Comment by GordonSmith** [comment]
I would want to sort both the headers and children in a specific order,  My instinct was to look for a "index" item in the markdown header section...

**Comment by jannikbuschke** [comment]
Yes this feature I also need myself. I will probably have a bit of time in the upcoming weeks, but cannot promise anything yet.

Feel free to open PRs. An index property on frontmatter/markdown sounds reasonable to me.

**Comment by wagnerjt** [comment]
Hmm. It might be too troublesome for maintaining and updating your indices if the sorting index is tucked away in the markdown's Metadata.
I can just imagine someone having even 7 children under a parent, and creating a new child for index 2. It'd be a pain to fish and do a +1 to all the following markdown files.

I was thinking about a simple js object, where the parent's name is the key, and the children is an array, listed in order for the sidebar. When the template gets generated, we can combine this object and the graphql response to perform the ordering, defaulting to alphabetic if there is none.

Thoughts?

**Comment by GordonSmith** [comment]
I don't have strong feelings on it, but one comment about the index suggestion is that you don't need to do sequential numbers, IOW:  100, 200, 300 would work equally well (and leave you plenty of room to insert a new one quickly...)

But your suggestion would work also open the door to having the same solution for the main (parent) items?

**Comment by mrpotatoes** [comment]
[I put this in my version](https://github.com/tct-gaming/tct-warframe-player-guide). To note mine is much more organized but I'm just giving the code

In any of your slugs in your markdown add an order to your slug (requires a server restart)
```md
---
title: Intro
root: '/docs'
parents: ['Farming']
order: 100
---
```

In SidebarContents.jsx change this
```jsx
// Add this function somewhere
const orderComparator = (menuA, menuB) => {
  let comparison = 0;
  if (menuA.order > menuB.order) {
    comparison = -1;
  } else if (menuA.order < menuB.order) {
    comparison = 1;
  }

  return comparison;
}

// Where the children are being sorted change it to this
item.children.sort(orderComparator)
```
Finally in the same file change `convertToTree` to this:
```js
const convertToTree = data => (
  constructTree(data.map(edge => ({
    path: edge.node.fields.slug,
    key: edge.node.id,
    title: edge.node.frontmatter.title,
    parents: edge.node.frontmatter.parents,
    order: edge.node.frontmatter.order || 0,
  })))
)

```

**Comment by mrpotatoes** [comment]
I also have ordering for the parents but I'm not a fan of how I'm doing that. I can give my code on that too but I think I will wait to see what comes of this repo next. I think it would actually be easier once the `Gatsby` version is updated. Cuz then we get the explorer tab in the `graphiql` explorer.

**Comment by jannikbuschke** [comment]
Why not make the children part of the parent structure directly?
Something like

```
navigation = [
{
  dir: 'parent',
  displayName: 'Introduction', 
  children: [ ... ]
]

```
